src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -77,18 +81,19 @@ public async Task<SecretWithProjectsListResponseModel> ListByOrganizationAsync([
     [HttpPost("organizations/{organizationId}/secrets")]
     public async Task<SecretResponseModel> CreateAsync([FromRoute] Guid organizationId, [FromBody] SecretCreateRequestModel createRequest)
     {
-        if (!_currentContext.AccessSecretsManager(organizationId))
+        if (createRequest.ProjectIds != null && createRequest.ProjectIds.Length > 1)
         {
-            throw new NotFoundException();
+            throw new BadRequestException();
         }
```

Comment:
This seems better suited as a validation in the request model.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2896/commits/54ac00026b19b8f95f894f7501e7f2cdcac68aa7